keras/callbacks.py

```diff
@@ -985,23 +985,74 @@ def on_epoch_end(self, epoch, logs=None):
 
 @keras_export("keras.callbacks.TerminateOnNaN")
 class TerminateOnNaN(Callback):
-    """Callback that terminates training when a NaN loss is encountered."""
+    """Callback that terminates training when a NaN loss is encountered.
 
-    def __init__(self):
+    Args:
+        check_freq: `'batch'` or `'epoch'` or integer. When using `'epoch'`,
+            checks the losses for NaN after every epoch. If using an integer,
+            let's say `1000`, losses will be checked every 1000 batches.
+            `'batch'` is a synonym for `1`, meaning that check will happen
+            every batch. Defaults to `'batch'`.
+    """
+
+    def __init__(self, check_freq="batch"):
         super().__init__()
+        self.check_freq = 1 if check_freq == "batch" else check_freq
+        if self.check_freq != "epoch" and not isinstance(self.check_freq, int):
+            raise ValueError(
+                f"Unrecognized check_freq: {self.check_freq}. "
+                "Expected 'batch', 'epoch' or integer"
+            )
+        self._batches_seen_since_last_check = 0
+        self._last_batch_seen = 0
         self._supports_tf_logs = True
 
-    def on_batch_end(self, batch, logs=None):
+    def _check_nan(self, batch=None, epoch=None, logs=None):
         logs = logs or {}
         loss = logs.get("loss")
+
         if loss is not None:
             loss = tf_utils.sync_to_numpy_or_python_type(loss)
             if np.isnan(loss) or np.isinf(loss):
-                io_utils.print_msg(
-                    f"Batch {batch}: Invalid loss, terminating training"
-                )
+                if batch is not None:
+                    io_utils.print_msg(
+                        f"Batch {batch}: Invalid loss, terminating training"
```

Comment:
Suggest: also print the epoch number to be more clear

Reply:
The epoch number is not passed down to the `on_batch_end()` hooks. The change has to happen deep inside the training loop.

Reply:
@rchao ignore my above comment, I included epoch number by following the design of ModelCheckpoint.